### DISCUSSION TITLE
How to programmatically create new content item version?

### DISCUSSION THREAD

**Discussion by tropcicstefan** [OrchardCMS/OrchardCore#7067] [Q&A] [answered]
I have content type Page that has two parts TitlePart and FlowPart. On publishing of content item of another type (Article) I would like to create new version of a Page with updated data. By version I mean new row in Document table(not updated Version column) with same ContentItemId and different ContentItemVersionId.
I have working example but it just seems wrong...

```
var originalPage = await _contentManager.GetAsync(pageCategoryIndex.PageContentItemId);
var newPage = await _contentManager.CloneAsync(originalPage);
newPage.ContentItemId = originalPage.ContentItemId;

newPage.Apply<FlowPart>(originalPage.As<FlowPart>());
newPage.Apply<TitlePart>(originalPage.As<TitlePart>());

//alter values in newPage

await _contentManager.UpdateAsync(newPage);
await _contentManager.RemoveAsync(originalPage);
newPage.Latest = true;
await _contentManager.PublishAsync(newPage);

```
So my question is 'what is the correct way to create new content item version?'

**Comment by Piedone** [comment, marked as answer]
Perhaps not the most intuitive but if you do 

```
var page = await _contentManager.GetAsync(pageCategoryIndex.PageContentItemId, VersionOptions.DraftRequired);
```

it'll create a draft version for that content item (if it's draftable). Then you can alter and publish it, creating a new published version. No need for the rest of the hoops you jump through, so no need for cloning/remove/manual latest.
